File: csharp/soal7_1.cs
/*
Soal        : 7
Modelsoal   : 2
*/

using System;

internal static class soal7_1{
    static int biaya(string kategori, string jalur, int bulan) {
        int dsp=0,spp=0;
        
        switch(jalur){
            case "SBMPTN":
                switch(kategori){
                    case "A":
                        dsp = 5000000;
                        spp = 500000;
                        break;
                    case "B":
                        dsp = 15000000;
                        spp = 1000000;
                        break;
                    case "C":
                        dsp = 30000000;
                        spp = 2000000;
                        break;
            } 
            break;
            case "SNMPTN":
                switch(kategori){
                    case "A":
                        dsp = 7000000;
                        spp = 500000;
                        break;
                    case "B":
                        dsp = 17000000;
                        spp = 1000000;
                        break;
                    case "C":
                        dsp = 35000000;
                        spp = 2000000;
                        break;
            }
            break;
            case "Mandiri":
                switch(kategori){
                    case "A":
                        dsp = 10000000;
                        spp = 1000000;
                        break;
                    case "B":
                        dsp = 25000000;
                        spp = 2000000;
                        break;
                    case "C":
                        dsp = 50000000;
                        spp = 3000000;
                        break;
            }
            break;
        }
        return (dsp+(spp*bulan));
    }
    static int input(string msg){
        Console.Write(msg);
        return Convert.ToInt32(Console.ReadLine());
    }
    private static void Main(){
        int id,ortu,bulan;
        string[]    nama  = {"Mira","Nina","Oemar","Pena"};
        string[]    jalur = {"SBMPTN","SNMPTN","Mandiri","SBMPTN"};
        string[]    alamat= {"Sawojajar","Kedung Kandang","Ijen","Dinoyo"};

        string kategori="";

        id = input("masukkan id mahasiswa       : ")-1;
        //id=2-1;
        if(id<0 || id> nama.GetLength(0)){
            Console.WriteLine("Invalid input");
            Environment.Exit(0);
        }
        ortu = input("masukkan pendapat ortu      : ");
        bulan = input("masukkan jumlah bulan spp   : ");
        //ortu=20000;
        //bulan=2

        if (ortu >10000000){
            kategori = "C";
        }
        else if (ortu >=2000000){
            kategori = "B";
        }
        else if (ortu <=2000000){
            kategori = "A";
        }

        Console.WriteLine(
        "\n=================================\n" +
        "Id mahasiswa                    : "+(id+1)+"\n"+
        "Nama mahasiswa                  : "+nama[id]+"\n"+
        "\n" +
        "Jalur masuk                     : "+jalur[id]+"\n"+
        "Kategori pemdapatan             : "+kategori+"\n"+
        "Jumlah biaya                    : "+biaya(kategori,jalur[id],bulan)+"\n"+
        "Alamat                          : "+alamat[id]+"\n"+
        "=================================");
    }
}

File: X/rust/XRPL_UKL1/soal7_1.rs
use std::io::{self, Write};

fn biaya(kategori: &str, jalur: &str, bulan: i32) -> i32 {
    let (mut dsp, mut spp) = (0, 0);

    match jalur {
        "SBMPTN" => match kategori {
            "A" => {
                dsp = 5_000_000;
                spp = 500_000;
            }
            "B" => {
                dsp = 15_000_000;
                spp = 1_000_000;
            }
            "C" => {
                dsp = 30_000_000;
                spp = 2_000_000;
            }
            _ => {}
        },
        "SNMPTN" => match kategori {
            "A" => {
                dsp = 7_000_000;
                spp = 500_000;
            }
            "B" => {
                dsp = 17_000_000;
                spp = 1_000_000;
            }
            "C" => {
                dsp = 35_000_000;
                spp = 2_000_000;
            }
            _ => {}
        },
        "Mandiri" => match kategori {
            "A" => {
                dsp = 10_000_000;
                spp = 1_000_000;
            }
            "B" => {
                dsp = 25_000_000;
                spp = 2_000_000;
            }
            "C" => {
                dsp = 50_000_000;
                spp = 3_000_000;
            }
            _ => {}
        },
        _ => {}
    }

    dsp + spp * bulan
}

fn input_int(msg: &str) -> i32 {
    print!("{}", msg);
    io::stdout().flush().unwrap();

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    input.trim().parse().unwrap()
}

fn main() {
    let nama = ["Mira", "Nina", "Oemar", "Pena"];
    let jalur = ["SBMPTN", "SNMPTN", "Mandiri", "SBMPTN"];
    let alamat = ["Sawojajar", "Kedung Kandang", "Ijen", "Dinoyo"];

    let id: usize = (input_int("Masukkan id mahasiswa: ") - 1) as usize;
    if id >= nama.len() {
        println!("Invalid input");
        return;
    }

    let ortu = input_int("Masukkan pendapatan ortu: ");
    let bulan = input_int("Masukkan jumlah bulan spp: ");

    let kategori = if ortu > 10000000 {
        "C"
    } else if ortu >= 2000000 {
        "B"
    } else {
        "A"
    };

    println!(
        "\n\
        =================================\n\
        Id mahasiswa                    : {}\n\
        Nama mahasiswa                  : {}\n\
        \n\
        Jalur masuk                     : {}\n\
        Kategori pendapatan             : {}\n\
        Jumlah biaya                    : {}\n\
        Alamat                          : {}\n\
        =================================\n",
        id + 1,
        nama[id],
        jalur[id],
        kategori,
        biaya(kategori, jalur[id], bulan),
        alamat[id]
    );
}
